genienlp/predict.py

```diff
@@ -184,15 +187,29 @@ def run(args, device):
                     raise OSError(f'{results_file_name} already exists')
 
             _, predictions, answers, contexts, _ = generate_with_model(model, it, numericalizer, task, args, prediction_file_name)
-                
+            
             if len(answers) > 0:
                 metrics_to_compute = task.metrics
                 if args.main_metric_only:
                     metrics_to_compute = [metrics_to_compute[0]]
                 metrics = calculate_and_reduce_metrics(predictions, answers, metrics_to_compute, args)
 
+                answers_wo_params = []
+                for a in answers:
+                    answers_wo_params.append(re.sub(quoted_pattern_maybe_space, '', a))
+                predictions_wo_params = [[] for _ in range(len(predictions))]
+                for i, preds in enumerate(predictions):
+                    for p in preds:
+                        predictions_wo_params[i].append(re.sub(quoted_pattern_maybe_space, '', p))
+
+                metrics_wo_params = calculate_and_reduce_metrics(predictions_wo_params, answers_wo_params, metrics_to_compute, args)
+                # rename em to sm
+                metrics_wo_params['sm'] = metrics_wo_params['em']
+                del metrics_wo_params['em']
```

Comment:
This is not the right metric. The right accuracy_without_params metric needs to consider all the entity tokens as well. You must use `genie evaluate-server` if you want consistent partial-accuracy metrics.

Reply:
This is for getting accuracy on datasets where entities don't match anymore. e.g. directly translated eval/ test sets without alignment in SPL.
I updated the implementation by porting the requoting functions from genie.

Reply:
I guess...